### PR TITLE
fixes #5, add a finalizer to fix deletion during pv protection

### LINKED ISSUE #5: PV Protection breaks deleting
If a user deletes a PV while it has a PVC with a pv protection finalizer the CSI never gets alerted to delete the volume when the pvc is deleted.

Related Issue: https://github.com/kubernetes-csi/external-provisioner/issues/195

It seems like https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/blob/master/controller/controller.go#L860 is the problem. 

Without a fix it is very easy for a user to delete a pv and break deleting via the CSI.

### PR DESCRIPTION
**Pull request by rmb938**
https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/issues/5

### CODE REVIEW

**Comment by rmb938** [comment]
It seems the tests are failing on getting the test-pod. Not sure what's going on any ideas?

File: controller/controller.go [hunk 1]
```diff
@@ -1138,6 +1155,33 @@ func (ctrl *ProvisionController) deleteVolumeOperation(volume *v1.PersistentVolu
...
 	}
 
+	if len(newVolume.ObjectMeta.Finalizers) > 0 {
+		// Remove external-provisioner finalizer
+
+		// need to get the pv again because the delete has updated the object with a deletion timestamp
+		newVolume, err := ctrl.client.CoreV1().PersistentVolumes().Get(volume.Name, metav1.GetOptions{})
+		if err != nil {
+			// If we get an error the volume is probably gone
+			return nil
```
Comment (wongma7) [inline controller/controller.go]: we can check for apierrs.IsNotFound explicitly here as well as we do for Update below

**Comment by wongma7** [comment]
thanks for figuring out the issue and this patch, don't worry about the test I am looking into it

**Comment by wongma7** [comment]
@rmb938 the finalizer name needs changing
"I1221 20:56:35.169970       1 controller.go:1075] provision "default/hostpath-pvc" class "example-hostpath": failed to save persistentvolume "pvc-da7cac17-0562-11e9-b559-b2cbe663ac83": PersistentVolume "pvc-da7cac17-0562-11e9-b559-b2cbe663ac83" is invalid: metadata.finalizers[0]: Invalid value: "external-provisioner.volume.kubernetes.io": name is neither a standard finalizer name nor is it fully qualified"

How it worked locally for me, I have no idea!

**Comment by rmb938** [comment]
Huh strange...

It seems finalizers have changed in 1.13 to have the syntax of label names. So I'll have to change it to something like `external-provisioner.volume.kubernetes.io/finalizer`

**Comment by wongma7** [comment]
/lgtm
/approve
thanks again & happy holidays.

feel free to squash if you like, I will re-add lgtm.

**Comment by rmb938** [comment]
@wongma7 Could we get this merged and released?

**Comment by wongma7** [comment]
/lgtm
yes, sorry for the delay, a release will be tagged today

**Pull request merged**